packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    vertical: PropTypes.bool,
+  }
+
+  static defaultProps: SliderProps = {
+    accessibility: sliderBehavior,
+    getA11yValueMessageOnChange: ({ value }) => String(value),
+    max: defaultPropValues.max,
+    min: defaultPropValues.min,
+    step: defaultPropValues.step,
+  }
+
+  static autoControlledProps = ['value']
+
+  getInitialAutoControlledState(): Partial<SliderState> {
+    return { value: defaultPropValues.value }
+  }
+
+  handleChange = (e: React.ChangeEvent) => {
+    const value = _.get(e, 'target.value')
+    _.invoke(this.props, 'onChange', e, { ...this.props, value })
+    this.trySetState({ value })
+  }
+
+  handleFocus = (e: React.FocusEvent) => {
+    this.setState({ isFromKeyboard: isFromKeyboard() })
+    _.invoke(this.props, 'onFocus', e, this.props)
+  }
+
+  handleMouseDown = (e: React.FocusEvent) => {
+    setWhatInputSource('mouse')
+    this.setState({ isFromKeyboard: false })
+    _.invoke(this.props, 'onMouseDown', e, this.props)
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    unhandledProps,
+  }: RenderResultConfig<SliderProps>) {
+    const { input, inputRef, step } = this.props
+    const [htmlInputProps, restProps] = partitionHTMLProps(unhandledProps)
+    const type = 'range'
+
+    const { min, max, value, valueAsPercentage } = processInputValues({
+      min: this.props.min,
+      max: this.props.max,
+      value: this.state.value || '',
+    })
+
+    // we need 2 wrappers around the slider rail, track, input and thumb slots to achieve correct component sizes
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...restProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        <div className={cx(Slider.slotClassNames.inputWrapper, classes.inputWrapper)}>
+          <span className={cx(Slider.slotClassNames.rail, classes.rail)} />
+          <span
+            className={cx(Slider.slotClassNames.track, classes.track)}
+            style={{ width: valueAsPercentage }}
+          />
+          <Ref
+            innerRef={(inputElement: HTMLElement) => {
+              handleRef(this.inputRef, inputElement)
+              handleRef(inputRef, inputElement)
+            }}
+          >
+            {Box.create(input || type, {
+              defaultProps: {
+                ...htmlInputProps,
+                ...accessibility.attributes.input,
+                className: Slider.slotClassNames.input,
+                as: 'input',
+                min,
+                max,
+                step,
+                type,
+                value,
+                onChange: this.handleChange,
+                onFocus: this.handleFocus,
+                onMouseDown: this.handleMouseDown,
```

Comment:
Shouldn't the handlers go in the `overrideProps`? This way those will be overriden if the user provides the same one.

Reply:
ooops... great catch, done